src/cargo/core/features.rs

```diff
@@ -620,7 +620,30 @@ impl Features {
     ) -> CargoResult<()> {
         let nightly_features_allowed = self.nightly_features_allowed;
         let Some((slot, feature)) = self.status(feature_name) else {
-            bail!("unknown cargo feature `{}`", feature_name)
+            let mut msg = format!(
+                "feature `{feature_name}` is not a valid unstable feature for Cargo.toml\n\n"
+            );
+
+            if feature_name.contains('_') {
+                let _ = writeln!(msg, "Feature names must use '-' instead of '_'.");
```

Comment:
Should we take this a step further and
- See if the name with `-` exists?
- Use `closest_msg` to correct for typos? (this one might require more work to do)

Reply:
Fully understand doing things incrementally, isolating controversy, etc.  As I mentioned with at least the second, I wasn't even sure how trivial it was to do.  Even the first might not be trivial.  I brought this up in case it was of interest especially since I feel like we too easily forget to apply `closest_msg` to our error messages when it can be a big help to users.